swap/swap_test.go

```diff
@@ -301,6 +307,161 @@ func TestRepeatedBookings(t *testing.T) {
 	verifyBookings(t, swap, append(bookings, mixedBookings...))
 }
 
+func TestNewSwapFailure(t *testing.T) {
```

Comment:
All tests should have a description of what they are doing. Sometimes the name is enough, but this time it is not really. 

One should not first have to read the test before understanding what it is doing, like it has been commented in the other test!

Reply:
Agree. Will change in about 30 minutes from now

Reply:
done

Reply:
...attempt**s**...

...whether it fails indeed....      or     ....whethere ther are indeed failures....  (watch double indeed also)

Reply:
done